core/lib/dal/src/transactions_dal.rs

```diff
@@ -2212,6 +2238,29 @@ impl TransactionsDal<'_, '_> {
         .fetch_optional(self.storage)
         .await
     }
+
+    pub async fn get_storage_tx_by_hash(
+        &mut self,
+        hash: H256,
+    ) -> DalResult<Option<StorageTransaction>> {
+        sqlx::query_as!(
+            StorageTransaction,
+            r#"
+            SELECT
+                *
+            FROM
+                transactions
+            WHERE
+                hash = $1
+            "#,
+            hash.as_bytes()
+        )
+        .map(Into::into)
+        .instrument("get_tx_error_by_hash")
```

Comment:
Nit: Mismatch between the method name / semantics and the name here. As we've discussed, I'd prefer if the method actually just loaded the error btw 🙃